Sporadic test failures in `test_name_resolve_cache`

### Description
I am seeing inconsistent failures in the above test when `get_icrs_coordinates` is not using the expected URL:

```python
    @pytest.mark.remote_data
    def test_name_resolve_cache(tmpdir):
        from astropy.utils.data import get_cached_urls
    
        target_name = "castor"
    
        temp_cache_dir = str(tmpdir.mkdir('cache'))
        with paths.set_temp_cache(temp_cache_dir, delete=True):
            assert len(get_cached_urls()) == 0
    
            icrs1 = get_icrs_coordinates(target_name, cache=True)
    
            urls = get_cached_urls()
            assert len(urls) == 1
>           assert 'http://cdsweb.u-strasbg.fr/cgi-bin/nph-sesame/' in urls[0]
E           AssertionError: assert 'http://cdsweb.u-strasbg.fr/cgi-bin/nph-sesame/' in 'http://vizier.cfa.harvard.edu/viz-bin/nph-sesame/A?castor'

/opt/lib/python3.9/site-packages/astropy/coordinates/tests/test_name_resolve.py:158: AssertionError
```
Apparently the function is somewhat randomly using the Vizier instead of the CDS mirror (this is from a European domain).

### System Details
macOS-10.14.6-x86_64-i386-64bit
Python 3.9.5 (default, May 14 2021, 02:39:23) 
[Clang 11.0.0 (clang-1100.0.33.17)]
Numpy 1.21.0rc2
astropy 4.3rc1
Scipy 1.6.3
Matplotlib 3.3.3

--- a/astropy/coordinates/tests/test_name_resolve.py
+++ b/astropy/coordinates/tests/test_name_resolve.py
@@ -155,7 +155,8 @@ def test_name_resolve_cache(tmpdir):
 
         urls = get_cached_urls()
         assert len(urls) == 1
-        assert urls[0].rstrip('A?castor') in sesame_url.get()
+        expected_urls = sesame_url.get()
+        assert any([urls[0].startswith(x) for x in expected_urls]), f'{urls[0]} not in {expected_urls}'
 
         # Try reloading coordinates, now should just reload cached data:
         with no_internet():

--- a/astropy/coordinates/tests/test_name_resolve.py
+++ b/astropy/coordinates/tests/test_name_resolve.py
@@ -155,7 +155,7 @@ def test_name_resolve_cache(tmpdir):
 
         urls = get_cached_urls()
         assert len(urls) == 1
-        assert 'http://cdsweb.u-strasbg.fr/cgi-bin/nph-sesame/' in urls[0]
+        assert urls[0].rstrip('A?castor') in sesame_url.get()
 
         # Try reloading coordinates, now should just reload cached data:
         with no_internet():

